### ISSUE TITLE
配置多个上下文，不调用 EnhanceLlamaIndexTool 的 run 方法

### ISSUE THREAD

**Issue by hzy970907** [langgenius/dify#407]
Dify version: Cloud | Self Host

## Steps To Reproduce

1.配置多个上下文

![image](https://github.com/langgenius/dify/assets/26761454/7a130978-5e2d-42f7-895c-06a57091c0df)

## The current behavior
不调用 EnhanceLlamaIndexTool 的 run 方法，route.destination 的值一直是 None。

![image](https://github.com/langgenius/dify/assets/26761454/f41a4c85-738e-41d5-8d80-8f59b99ccbf8)

## The expected behavior
调用 EnhanceLlamaIndexTool 的 run 方法。

**Comment by takatost** [comment]
This requires your dataset description to be able to summarize the content of the dataset so that RouteChain can determine if your query is related to the dataset.

**Comment by hzy970907** [comment]
![image](https://github.com/langgenius/dify/assets/26761454/b4cb00f9-1112-46d9-8088-19667f6c4cfa)

<br/>
dataset description 我用的自动生成的，没有动过，dify 目前不支持配置多个上下文？
只要我配置多个上下文，EnhanceLlamaIndexTool 的 run 方法就不执行。Datasets&Index 功能就失效。

**Comment by takatost** [comment]
you need to manually modify the description of your dataset, the auto-generated one is just a file name

**Comment by hzy970907** [comment]
配置多个上下文，EnhanceLlamaIndexTool 的 run 方法不执行，与 dataset description 有关？

**Comment by takatost** [comment]
Yep, we'll choose which dataset to use based on the user's question and the dataset description.
